src/Analyser/MutatingScope.php

```diff
@@ -2954,7 +2961,11 @@ private function enterFunctionLike(
 		foreach (ParametersAcceptorSelector::selectSingle($functionReflection->getVariants())->getParameters() as $parameter) {
 			$parameterType = $parameter->getType();
 			if ($parameter->isVariadic()) {
-				$parameterType = new ArrayType(new IntegerType(), $parameterType);
+				if ($this->phpVersion->supportsNamedArguments()) {
+					$parameterType = new ArrayType(new BenevolentUnionType([new IntegerType(), new StringType()]), $parameterType);
```

Comment:
I'm pondering if it should be a benevolent union type or not. Maybe we want a normal union type so that the user always finds out about a possible bug?